### ISSUE TITLE
IDE0008 not configurable through .editorconfig

### ISSUE THREAD

**Issue by BastianMattes** [dotnet/roslyn#57545]
**Version Used**: 

Visual Studio 2022 (17.0.0 Preview 7.0)

**Steps to Reproduce**:

1. Create a console application with the following Program.cs:

```
using System.IO;

var _ = new StringReader("T");`
```

The project looks like this:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>disable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <AnalysisLevel>preview</AnalysisLevel>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.CodeStyle" Version="4.0.0-6.final">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

2. Add a .editorconfig with the following content:

```
[*.cs]

# Default severity for analyzer diagnostics with category 'Style'
dotnet_analyzer_diagnostic.severity = warning

csharp_style_var_for_built_in_types=false:suggestion
csharp_style_var_when_type_is_apparent=false:suggestion
csharp_style_var_elsewhere=false:suggestion
```

3. Build the project

**Expected Behavior**:

No warnings are shown in the build output.
Visual Studio shows a suggestion to use the explicit type instead of 'var'.

**Actual Behavior**:

The build output shows:

> [...]\ConsoleApp1\Program.cs(3,1,3,4): warning IDE0008: Use explicit type instead of 'var'

Visual Studio does not show any suggestions.

**Comment by sharwell** [comment]
I believe the analyzer is behaving as expected: the configuration of `dotnet_analyzer_diagnostic.severity` is applied to diagnostics produced by the analyzer, and there is no use of `dotnet_diagnostic.IDE0008.severity` in the file to apply a more specific value to that diagnostic.

**Comment by BastianMattes** [comment]
Thank you for clarifying.
